pkg/cloud/metadata/ec2.go

```diff
@@ -88,7 +88,7 @@ func EC2MetadataInstanceInfo(svc EC2Metadata, regionFromSession string) (*Metada
 	if err != nil {
 		return nil, fmt.Errorf("could not read ENIs metadata content: %w", err)
 	}
-	attachedENIs := strings.Count(string(enis), "\n")
+	attachedENIs := util.CountMACAddresses(string(enis))
```

Comment:
Idea: suggest adding a log message with this information, it could be useful for debugging. Since this is only logged during driver startup the default v-level is fine, but 4 or 5 sounds more appropriate.

```
klog.InfoS("Number of attached ENIs", "attachedENIs", attachedENIs)
```